flink-connectors/flink-connector-kafka-0.9/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumer09.java

```diff
@@ -208,13 +208,12 @@ public FlinkKafkaConsumer09(List<String> topics, KeyedDeserializationSchema<T> d
 				if (partitionsForTopic != null) {
 					partitions.addAll(convertToFlinkKafkaTopicPartition(partitionsForTopic));
 				}
+				else{
+					throw new RuntimeException("Unable to retrieve any partitions for the requested topics " + topic);
```

Comment:
I think throwing an exception here is not a good idea.
If you have 10 topics, and only one of them doesn't have any partitions, the consumer won't start anymore.
Also the exception message is not accurate. It says Flink can not retrieve ANY partitions for the requested topicS. However, its just the partitions of one topic.

I suggest to only log a INFO-level message in that case.